dask/dataframe/io/parquet.py

```diff
@@ -576,17 +584,28 @@ def _get_pyarrow_dtypes(schema):
 def _read_pyarrow_parquet_piece(fs, piece, columns, index_cols, is_series,
                                 partitions):
     with fs.open(piece.path, mode='rb') as f:
-        table = piece.read(columns=columns, partitions=partitions,
+        table = piece.read(columns=index_cols + columns,
+                           partitions=partitions,
                            use_pandas_metadata=True,
                            file=f)
     df = table.to_pandas()
-    if (index_cols and df.index.name is None and
-            len(df.columns.intersection(index_cols))):
+    has_index = (df.index.names != [None] or
+                 type(df.index) != pd.RangeIndex)
```

Comment:
I don't know pyarrow - is it possible to have a legitimate RangeIndex (not starting from zero)?

Reply:
A `RangeIndex` only occurs when not read from data.

Reply:
Should this be an instance check rather than a direct type comparison? If not, should probably also use `is not` instead of `!=`

Reply:
Simplified to just `not isinstance(df.index, pd.RangeIndex)`